api/solarperformanceinsight_api/models.py

```diff
@@ -548,3 +552,371 @@ class UserInfo(StoredObject):
     """Information about the current user"""
 
     auth0_id: str = Field(..., description="User ID from Auth 0")
+
+
+class JobTimeindex(BaseModel):
+    """Parameters for a time index that all data uploads must conform to.
+    Data is assumed to time-averaged and closed and labeled at the left endpoint, i.e.
+    a datapoint at 23:00 of data with a 1 hour time step is assumed be the
+    average of data from 23:00 to 23:59.
+    """
+
+    start: dt.datetime = Field(
+        ..., description="Start of the time range that data will be uploaded for"
+    )
+    end: dt.datetime = Field(
+        ...,
+        description="End (exclusive) of the time range that data will be uploaded for",
+    )
+    step: dt.timedelta = Field(
+        ..., description="Time step between each data point from 1 to 60 minutes"
+    )
+    timezone: Optional[str] = Field(
+        ...,
+        description="Timezone data will be converted to before computation. "
+        "Unlocalized data will be localized to this timezone. If this is null, "
```

Comment:
```suggestion
        "Unlocalized data will be localized to this timezone. If timezone is null, "
```

Probably should include an example string, e.g. 'Etc/GMT+5'. I usually have to look up the pytz timezone names.